src/main.cpp

```diff
@@ -679,9 +679,6 @@ bool IsStandardTx(const CTransaction& tx, string& reason)
         else if ((whichType == TX_MULTISIG) && (!fIsBareMultisigStd)) {
             reason = "bare-multisig";
             return false;
-        } else if (txout.IsDust(::minRelayTxFee)) {
```

Comment:
A txout with 0 balance has no use.  It's just bloat.  The exception is if the txout with 0 balance has an OP_RETURN  which may contain useful info.

So in order to avoid bloat from honest non-miners, I think we should at least retain the behaviour that a standard tx can only have one txout with 0 balance and it must be an OP_RETURN.

This doesn't stop an attacker creating txouts with 1 Zatoshi, which is basically free.

Also a miner could collude to include non-standard transactions with many 0 txouts in order to increase the size of the blockchain.

Can this PR include ways to prevent this from happening?  Ping @nathan-at-least